app/assets/javascripts/modules/testsByCategoryWidget.js

```diff
@@ -0,0 +1,166 @@
+angular.module('probedock.testsByCategoryWidget', [ 'probedock.api' ])
+  .directive('testsByCategoryWidget', function() {
+    return {
+      restrict: 'E',
+      controller: 'TestsByCategoryBarsCtrl',
+      templateUrl: '/templates/tests-by-category-widget.html',
+      scope: {
+        organization: '=',
+        project: '=',
+        nbCategories: '=?'
```

Comment:
Small thing: in the case of the by day/week widgets, the number you pass to the directive is always the number of days/weeks that will be displayed, so it makes sense to have `nbDays` as a parameter name. But in this case, there may be fewer categories displayed, so it's a maximum. I would call this `maxNbCategories`. And maybe add a comment (on the directive or on the scope parameter) to explain that an "Other" will be displayed and whether it is included in this number.